open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,512 @@
+     *
+     * @param submissionJson
+     * @return
+     */
+    public static JSONObject validateRiskParameter(JSONObject submissionJson) {
+
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+
+        //Validate RISK
+        if (submissionJson.containsKey(ConsentExtensionConstants.RISK)) {
+
+            Object dataObject = submissionJson.get(ConsentExtensionConstants.RISK);
+            // Check if the risk is valid JSON Object
+            if (!isValidJSONObject(dataObject)) {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                        ErrorConstants.RISK_NOT_JSON_ERROR);
+            }
+        } else {
+            log.error(ErrorConstants.RISK_NOT_FOUND);
+            return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                    ErrorConstants.RISK_NOT_FOUND);
+        }
+        return validationResult;
+    }
+
+    /**
+     * Checks if the given Object is a JSONObject and the JSONObject is non-empty , and it is an instance of a string.
+     *
+     * @param value The Object to be validated.
+     * @return true if the object is a non-null and non-empty JSONObject.
+     */
+    public static boolean isValidString(Object value) {
+        return value instanceof String;
+    }
+
+    /**
+     * Validates initiation parameter in the submission data.
+     *
+     * @param submissionData The JSONObject containing submission data.
+     * @return A JSONObject indicating the validation result.
+     */
+    public static JSONObject validateInitiationParameter(JSONObject submissionData) {
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+
+        if (submissionData.containsKey(ConsentExtensionConstants.INITIATION)) {
+
+            Object dataObject = submissionData.get(ConsentExtensionConstants.INITIATION);
+
+            if (!isValidJSONObject(dataObject)) {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                        ErrorConstants.INITIATION_NOT_JSON);
+            }
+        } else {
+            log.error(ErrorConstants.INITIATION_NOT_FOUND);
+            return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                    ErrorConstants.INITIATION_NOT_FOUND);
+        }
+
+        return validationResult;
+    }
+
+    /**
+     * Validates instruction parameter in the submission data.
+     *
+     * @param submissionData The JSONObject containing submission data.
+     * @return A JSONObject indicating the validation result.
+     */
+    public static JSONObject validateInstructionParameter(JSONObject submissionData) {
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
```

Comment:
Usually it's not a good practice to set the validation result as success , Because there are many scenarios that validation could fail. Therefor it's good to first assume validation if failed, and only the checks are passed we set the validation to true.

Reply:
Fixed in [a613632]